book/running-wallaroo/running-wallaroo.md

```diff
@@ -0,0 +1,93 @@
+# Running Wallaroo
+
+A Wallaroo application is distributed over one or more Wallaroo processes, which we call "workers". Any Wallaroo cluster will have one worker designated as the "cluster initializer", a temporary role played during cluster initialization. 
+
+Workers communicate with each other over two channels: a control channel and a data channel. They send metrics data over TCP to an external metrics address. And they optionally listen for messages from external systems over an external channel.  
+
+Depending on the type of sources and sinks used in your application, you might need to provide more configuration information.  For example, a TCP source requires a network address so that it can listen for incoming data.  And a TCP sink requires a target address where it can send output data.
+
+In this document, we'll discuss most of the command line arguments that can be passed to a Wallaroo binary.  For a complete list, see [Wallaroo Command Line Options](/book/running-wallaroo/wallaroo-command-line-options.md).
+
+## Single Worker Setup
+
+If you are starting up a Wallaroo cluster with only one worker, then that worker must be configured as the "cluster initializer" by passing the `--cluster-initializer` flag. You must also specify a target address for metrics data using the `--metrics` parameter. And you must specify its control and data addresses via the `--control` and `--data` command line parameters (these will be used if more workers are added to the cluster later). 
+
+If your application uses a TCP source, then you must specify a TCP input address via `--in`. Likewise, if your application uses a TCP sink, then you must specify a TCP output address via `--out`. In what follows, we'll be using the Celsius Converter example app which uses a TCP source and sink.
+
+If you are using the Go or Pony APIs, then you will run a binary called `celsius` in order to run the Celsius Converter app. If you are using the Python API, then you will run [Machida](book/python/intro.md), passing in the application name via `--application-module`. And since Machida is single-threaded, you must pass in `--ponythreads 1` (otherwise the cluster will crash).
+
+Putting this all together, to run the Celsius Converter app, you would run the following command:
```

Comment:
It might be worthwhile to provide links to the Go and Python versions of the "Celsius Converter" app so there is absolutely no confusion about what we're talking about here.

Reply:
done